lib/engine/game/g_steam_over_holland/step/buy_train.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/buy_train'
+
+module Engine
+  module Game
+    module GSteamOverHolland
+      module Step
+        class BuyTrain < Engine::Step::BuyTrain
+          def actions(entity)
+            return [] if entity != current_entity
+
+            return ['buy_train'] if can_afford_needed_train?(entity)
+            return ['sell_shares'] if entity == current_entity&.owner && can_ebuy_sell_shares?(current_entity)
+            return %w[sell_shares buy_train pass] if president_may_contribute?(entity)
+            return %w[buy_train pass] if can_buy_train?(entity)
+
+            []
+          end
+
+          def pass_description
+            if can_close_corp?(current_entity)
+              "Close #{current_entity.name}"
+            else
+              @acted ? 'Done (Trains)' : 'Skip (Trains)'
+            end
+          end
+
+          def process_pass(action)
+            entity = action.entity
+            return super unless can_close_corp?(entity)
+
+            @game.close_corporation(entity)
+          end
+
+          def must_sell_shares?(corporation)
+            return false if corporation.cash >= @game.depot.min_depot_price || !corporation.trains.empty?
+
+            can_issue?(corporation)
+          end
+
+          def can_issue?(corporation)
+            return false unless corporation.corporation?
+            return false if @round.issued_shares[corporation]
+
+            @game.issuable_shares(corporation).any?
```

Comment:
Don't call `any?` or `none?` without a block. Use `empty?` instead.
```suggestion
            !@game.issuable_shares(corporation).empty?
```

Reply:
done